.travis.yml

```diff
@@ -23,35 +24,58 @@ env:
    - IOS_SDK=iphonesimulator12.1
    - TVOS_SDK=appletvsimulator12.1
 
-   matrix:
-   - DESTINATION="OS=11.4,name=iPhone X" SDK="$IOS_SDK" BUILD_EXAMPLE="YES"
-
-   - DESTINATION="OS=12.1,name=iPhone X" SDK="$IOS_SDK" BUILD_EXAMPLE="YES"
-
-   - DESTINATION="OS=12.1,name=Apple TV" SDK="$TVOS_SDK" BUILD_EXAMPLE="NO"
-
 before_install:
   - pip3 install proselint
   - gem update --system
   - gem install bundler
 
-script:
-- set -o pipefail
-- xcodebuild -version -sdk
+jobs:
+  include:
+    - &test
+      stage: test
+      name: iOS iPhone X 11.4
+      env: DESTINATION="OS=11.4,name=iPhone X" SDK="$IOS_SDK" BUILD_EXAMPLE="YES"
+      script:
+        - set -o pipefail
+        - xcodebuild -version -sdk
+        - xcodebuild clean test -project "$PROJECT" -scheme JSQDataSourcesKit -sdk "$SDK" -destination "$DESTINATION" -enableCodeCoverage YES -configuration Debug ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO | bundle exec xcpretty -c;
+      after_success:
+        - bundle exec danger
+        - bash <(curl -s https://codecov.io/bash)
 
-- if [ $BUILD_EXAMPLE == "YES" ]; then
-      xcodebuild build analyze -project Example/Example.xcodeproj -scheme Example -sdk "$SDK" -destination "$DESTINATION" ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO | bundle exec xcpretty -c;
-  fi
+    - <<: *test
+      name: iOS iPhone X 12.1
+      env: DESTINATION="OS=12.1,name=iPhone X" SDK="$IOS_SDK" BUILD_EXAMPLE="YES"
 
-- xcodebuild analyze test -project "$PROJECT" -scheme JSQDataSourcesKit -sdk "$SDK" -destination "$DESTINATION" -enableCodeCoverage YES -configuration Debug ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO | bundle exec xcpretty -c;
+    - <<: *test
+      name: Apple TV 12.1
+      env: DESTINATION="OS=12.1,name=Apple TV" SDK="$TVOS_SDK" BUILD_EXAMPLE="NO"
 
-- bundle exec danger
+    - &build
+      stage: build
+      name: iOS iPhone X 11.4
+      env: DESTINATION="OS=11.4,name=iPhone X" SDK="$IOS_SDK" BUILD_EXAMPLE="YES"
+      script:
+        - set -o pipefail
+        - xcodebuild -version -sdk
+        - xcodebuild clean build -project Example/Example.xcodeproj -scheme Example -sdk "$SDK" -destination "$DESTINATION" -configuration Debug ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO | bundle exec xcpretty -c;
 
-after_success:
-- bash <(curl -s https://codecov.io/bash)
+    - <<: *build
+      name: iOS iPhone X 12.1
+      env: DESTINATION="OS=12.1,name=iPhone X" SDK="$IOS_SDK" BUILD_EXAMPLE="YES"
 
-jobs:
-  include:
-  - stage: lint
-    name: "pod lint"
-    script: bundle exec pod lib lint
+    - stage: lint
+      name: "pod lint"
+      script: bundle exec pod lib lint
+
+    - stage: carthage
+      osx_image: xcode10.1
+      name: Carthage
+      os: osx
+      script: carthage build --no-skip-current --configuration Release
+
+stages:
+  - test
+  - build
+  - lint
+  - carthage
```

Comment:
💯

Reply:
One question:  
Do you want to conditionally build any stage(s)?  (i.e. only run ```carthage``` if branch == master && !PR).

Reply:
meh, i think it's fine to always run. it shouldn't take long

(and i'd rather catch bugs early)